server/src/test/java/keywhiz/service/providers/ClientAuthFactoryTest.java

```diff
@@ -319,4 +366,18 @@ public void rejectsXfcc_missingClientCert() {
 
     factory.provide(request, httpServletRequest);
   }
+
+  @Test(expected = NotAuthorizedException.class)
+  public void rejectsXfcc_customSpiffeIdHeader_clientNotPresent() {
+    when(httpServletRequest.getLocalPort()).thenReturn(xfccAllowedPort);
+    when(request.getRequestHeader(ClientAuthFactory.XFCC_HEADER_NAME)).thenReturn(
+        List.of(xfccHeader));
+    when(securityContext.getUserPrincipal()).thenReturn(xfccPrincipal);
+
+    when(xfccSourceConfig.callerSpiffeIdHeader()).thenReturn(callerSpiffeHeaderName);
+    when(request.getRequestHeader(callerSpiffeHeaderName)).thenReturn(
+        List.of(clientSpiffeNotAllowed));
```

Comment:
nit: since the issue here is that the client doesn't exist and we didn't configure Keywhiz to create missing clients, I'd recommend naming this variable `nonexistentClientSpiffe` or something similar. ("not allowed" makes me think the client might exist, but be forbidden to do whatever I tried to do.)